zerver/tests/test_subs.py

```diff
@@ -2874,13 +2874,75 @@ def test_remove_user_from_multiple_streams(self) -> None:
                             "principals": orjson.dumps(principals).decode(),
                         },
                     )
-        self.assert_length(cache_tries, 4)
+        self.assert_length(cache_tries, 13)
 
         if result.status_code not in [400]:
             for name in stream_names:
                 users_subbed_to_streams = self.users_subscribed_to_stream(name, subbed_user.realm)
             self.assertNotIn(subbed_user, users_subbed_to_streams)
 
+    def test_remove_subscription_user_notification(self) -> None:
+        admin = self.example_user("iago")
+        self.login_user(admin)
+        subbed_user = self.example_user("cordelia")
+
+        # Set up the stream.
+        stream_names = ["hümbüǵ"]
+        for name in stream_names:
+            self.subscribe(subbed_user, name)
+
+        # Set up the principal to be unsubscribed.
+        principals = [subbed_user.id]
+
+        self.client_delete(
+            "/json/users/me/subscriptions",
+            {
+                "subscriptions": orjson.dumps(stream_names).decode(),
+                "principals": orjson.dumps(principals).decode(),
+            },
+        )
+
+        msg = "@_**Iago|11** unsubscribed you from the stream #**hümbüǵ**."
+
+        GetSubscribersTest.assert_user_got_bot_message(GetSubscribersTest(), subbed_user, msg)
+
+    def test_remove_multiple_subscription_user_notification(self) -> None:
+        admin = self.example_user("iago")
+        self.login_user(admin)
+        subbed_user = self.example_user("cordelia")
+
+        # Set up the stream.
+        stream_names = ["hümbüǵ"] + [f"Stream {n}" for n in range(10)]
+        for name in stream_names:
+            self.subscribe(subbed_user, name)
+
+        # Set up the principal to be unsubscribed.
+        principals = [subbed_user.id]
+
+        self.client_delete(
+            "/json/users/me/subscriptions",
+            {
+                "subscriptions": orjson.dumps(stream_names).decode(),
+                "principals": orjson.dumps(principals).decode(),
+            },
+        )
+
+        msg = """@_**Iago|11** unsubscribed you from the following streams:
+        *#**Stream0**
+        *#**Stream1**
+        *#**Stream2**
+        *#**Stream3**
+        *#**Stream4**
+        *#**Stream5**
+        *#**Stream6**
+        *#**Stream7**
+        *#**Stream8**
+        *#**Stream9**
+        *#**hümbüǵ**
+        """
+
+        GetSubscribersTest.assert_user_got_bot_message(GetSubscribersTest(), subbed_user, msg)
```

Comment:
I think rather than adding these two new tests, you should just modify the existing tests that remove users from streams to assert these notifications were sent and verify their content. Since it's an unconditional part of removing a subscriber, at least for now, there's no reason to repeat the test setup to verify the behavior.